src/lightning_app/cli/cmd_clusters.py

```diff
@@ -198,6 +203,33 @@ def delete(self, cluster_id: str, force: bool = False, do_async: bool = False) -
             )
             click.confirm("Do you want to continue?", abort=True)
 
+        else:
+            apps = _list_apps(self.api_client, cluster_id=cluster_id, phase_in=[V1LightningappInstanceState.RUNNING])
+            if apps:
+                raise click.ClickException(
+                    dedent(
+                        """\
+                        To delete the cluster, you must first delete the apps running on it.
+                        Use the following commands to delete the apps, then delete the cluster again:
+
+                        """
+                    )
+                    + "\n".join([f"\tlightning delete app {app.name} --cluster-id {cluster_id}" for app in apps])
+                )
+
+            if _list_apps(self.api_client, cluster_id=cluster_id, phase_not_in=[V1LightningappInstanceState.DELETED]):
```

Comment:
Question: Won't DELETED instance not be listed ?

Reply:
If `phase_not_in` and `phase_in` are not provided the default behaviour is to return all instances that are not deleted.

I could change the query to use this default behaviour, but it doesn't hurt to be explicit either.